server/rest/src/test/java/org/infinispan/rest/cachemanager/RestCacheManagerTest.java

```diff
@@ -60,17 +60,14 @@ public void shouldReuseEncodedCaches() {
       // Verify they are stored internally
       assertEquals(knownCaches.size(), 4);
 
-
-      // Requesting again with same media, or with same media but different parameters should reuse internal instance
+      // Requesting again with same media but different parameters should not reuse internal instance
       Mockito.reset(embeddedCacheManager);
-      restCacheManager.getCache("cache2", MediaType.MATCH_ALL, MediaType.TEXT_PLAIN, request);
       restCacheManager.getCache("cache2", MediaType.MATCH_ALL, MediaType.fromString("text/plain; charset=UTF-8"), request);
       restCacheManager.getCache("cache2", MediaType.MATCH_ALL, MediaType.fromString("text/plain; charset=SHIFT-JIS"), request);
 
-      assertEquals(knownCaches.keySet().size(), 4);
+      assertEquals(knownCaches.keySet().size(), 6);
```

Comment:
We should also verify that using a media type list with `q` parameters does not create a new encoded cache.

Reply:
I decided to take into account every parameter form the MediaType to associated it with an encoded cache, so they are all available to the transcoders